pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -243,19 +244,27 @@ func (c *Cluster) createSelfSignedCert() (bool, error) {
 			exitCode, parsed := c.exitCode(err)
 			if parsed {
 				if exitCode == certAlreadyConfiguredErrorCode {
```

Comment:
@alfonsomthd What is the proper way to detect if the self-signed cert is already created for the dashboard? When Rook is calling `ceph dashboard created-self-signed-cert`, it is returning success whether or not the cert already exists. We need to detect the difference between the cert already existing, and when it is newly created so we know whether to restart the module. In the past, this command returned error code `5` if it already existed, but ceph is no longer returning that error. Is there another command we can call to detect if it already exists?

Reply:
@travisn According to the [docs](https://docs.ceph.com/en/latest/mgr/dashboard/#id2), you should verify that by retrieving the corresponding config keys:

> Check if the SSL/TSL support is enabled:
> 
> `ceph config get mgr mgr/dashboard/ssl`
> If the command returns `true`, verify a certificate exists by:
> 
> `ceph config-key get mgr/dashboard/crt`
> and:
> `ceph config-key get mgr/dashboard/key`

@nizamial09 @avanthakkar @cloudbehl please confirm this (it's been a long time since my last dashboard contribution :stuck_out_tongue: )

Reply:
Perfect thanks, I'll take a look at that doc, I missed it.

Reply:
@alfonsomthd  It's working perfectly now with those checks, thanks